modules/rosa-list-objects.adoc

```diff
@@ -0,0 +1,443 @@
+
+|debug
+|Enables debug mode.
+
+|--profile
+|Uses a specific AWS profile (string) from your credentials file.
+
+|-v, --v level
+|Log level for V logs.
+|===
+
+.Example
+List all identity providers (IDPs) for a cluster named 'mycluster'.
+
+[source,terminal]
+----
+$ rosa list idps --cluster=mycluster
+----
+
+[id="rosa-list-ingresses_{context}"]
+== list ingresses
+
+Lists all API and ingress endpoints for a cluster.
+
+.Syntax
+[source,terminal]
+----
+$ rosa list ingresses --cluster=mycluster [arguments]
+----
+
+.Arguments
+|===
+|Option |Definition
+
+|-c, --cluster
+|(Required) The name or ID (string) of the cluster for which the IDPs will be listed.
+|===
+
+.Optional arguments inherited from parent commands
+|===
+|Option |Definition
+
+|-h, --help
+|Shows help for this command.
+
+|debug
+|Enables debug mode.
+
+|--profile
+|Uses a specific AWS profile (string) from your credentials file.
+
+|-v, --v level
+|Log level for V logs.
+|===
+
+.Example
+List all API and ingress endpoints for a cluster named 'mycluster'.
+
+[source,terminal]
+----
+$ rosa list ingresses --cluster=mycluster
+----
+
+[id="rosa-list-machinepools_{context}"]
+== list machinepools
+
+Lists the machine pools configured on a cluster.
+
+.Syntax
+[source,terminal]
+----
+$ rosa list machinepools [arguments]
+----
+
+.Arguments
+|===
+|Option |Definition
+
+|-c, --cluster
+|(Required) The name or ID (string) of the cluster for which the machine pools will be listed.
+|===
+
+.Optional arguments inherited from parent commands
+|===
+|Option |Definition
+
+|-h, --help
+|Shows help for this command.
+
+|debug
+|Enables debug mode.
+
+|--profile
+|Uses a specific AWS profile (string) from your credentials file.
+
+|-v, --v level
+|Log level for V logs.
+|===
+
+.Example
+List all of the machine pools on a cluster named 'mycluster'.
+
+[source,terminal]
+----
+$ rosa list machinepools --cluster=mycluster
+----
+
+[id="rosa-list-regions_{context}"]
+== list regions
+
+Lists all of the available regions for the current AWS account.
+
+.Syntax
+[source,terminal]
+----
+$ rosa list regions [arguments]
+----
+
+.Arguments
+|===
+|Option |Definition
+
+|--multi-az
+|Lists regions that support for multiple availability zones.
+|===
+
+.Optional arguments inherited from parent commands
+|===
+|Option |Definition
+
+|-h, --help
+|Shows help for this command.
+
+|debug
+|Enables debug mode.
+
+|--profile
+|Uses a specific AWS profile (string) from your credentials file.
+
+|-v, --v level
+|Log level for V logs.
+|===
+
+.Example
+List all of the available regions.
+
+[source,terminal]
+----
+$ rosa list regions
+----
+
+[id="rosa-list-upgrades_{context}"]
+== list upgrades
+
+Lists all available and scheduled cluster version upgrades.
+
+.Syntax
+[source,terminal]
+----
+$ rosa list upgrades [arguments]
+----
+
+.Arguments
+|===
+|Option |Definition
+
+|-c, --cluster
+|(Required) The name or ID (string) of the cluster for which the available upgrades will be listed.
```

Comment:
It's okay to end a sentence with "for" to avoid awkward wording:

```
The name or ID (string) of the cluster to list the available upgrades for.
```

Reply:
Done.